runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/OrchestratorRequestType.java

```diff
@@ -0,0 +1,53 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * The type of requests that can be made to the Orchestrator Service.
+ * 
+ * @author Maithem
+ */
+@AllArgsConstructor
+public enum OrchestratorRequestType {
+
+    /**
+     * Query a workflow id
+     */
+    QUERY(0),
+
+    /**
+     * Add a new node to the cluster
+     */
+    ADD_NODE(1);
+
+    @Getter
+    public int type;
+
+    /**
+     * Create a workflow instance from an orchestrator request
+     *
+     * @param request Orchestrator request
+     * @return Workflow instance
+     */
+    public IWorkflow getWorkflow(OrchestratorRequest request) {
```

Comment:
Is it possible to tease the workflow factory out of the request type. It is just easier for me to understand the code.